src/netif.rs

```diff
@@ -26,6 +26,7 @@ use crate::private::mutex;
 pub enum NetifStack {
     /// Station mode (WiFi client)
     Sta,
+    #[cfg(esp_idf_softap_support)]
```

Comment:
Where is this defined? The PR is currently breaking the compilation of the examples, because this `cfg` is actually not defined by default?

Reply:
Not sure it is defined by default, but it is used in the C side of the sdk. How shall I address the default value?

Reply:
CONFIG_ESP_WIFI_SOFTAP_SUPPORT
https://docs.espressif.com/projects/esp-idf/en/v5.1.3/esp32/api-reference/kconfig.html?highlight=softap_support#config-esp-wifi-softap-support

Reply:
Then the `cfg` on the Rust side needs to be named `esp_idf_esp_wifi_softap_support`.

Reply:
I'll update the PR, though unexpectedly this works in my local branch; enabling/disabling has the expected effect

Reply:
> I'll update the PR, though unexpectedly this works in my local branch; enabling/disabling has the expected effect

I do not understand. 
`esp_idf_softap_support` definitely does **not** work, and is incorrect, because `CONFIG_XYZ` ESP IDF configs are translated by `esp-idf-sys` into `cfg(esp_idf_<the-esp-idf-config-lowercased-and-with-stripped-CONFIG_-prefix>)`.

Or else the Wifi examples would've compiled, because `CONFIG_ESP_WIFI_SOFTAP_SUPPORT` - as per the docu - is enabled by default.

Reply:
Btw, you also need to update the failing examples, not to be compiled when this config is not defined.